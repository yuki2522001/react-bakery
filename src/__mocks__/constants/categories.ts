import { CategoryProps } from "@common-types/category";

const CATEGORY_MOCKING_LIST: CategoryProps[] = [
  {
    id: "1",
    name: "Cake"
  },
  {
    id: "2",
    name: "Breads"
  }
]

export { CATEGORY_MOCKING_LIST };
