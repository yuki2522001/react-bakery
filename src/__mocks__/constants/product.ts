import { Product } from "@common-types/product";
import url from "../../assets/images/products/black-forest-cake.jpg";

const PRODUCT_MOCKING: Product = {
  id: "1",
  categoryId: "1651924356334",
  name: "Cheese cake",
  price: 20,
  quantity: 5,
  description: "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium.",
  images: [
    url
  ]
}

const PRODUCT_MOCKING_LIST: Product[] = [
  {
    id: "1",
    categoryId: "1651924356334",
    name: "Cheese cake",
    price: 20,
    quantity: 5,
    description: "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium.",
    images: [
      "/src/assets/images/products/cheese-cake.jpg",
      "/src/assets/images/products/cheese-cake.jpg",
      "/src/assets/images/products/cheese-cake.jpg"
    ]
  },
  {
    id: "2",
    categoryId: "1651924356334",
    name: "Amaretti Cookie",
    price: 10,
    quantity: 5,
    description: "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium.",
    images: [
      "/src/assets/images/products/cheese-cake.jpg",
      "/src/assets/images/products/cheese-cake.jpg",
      "/src/assets/images/products/cheese-cake.jpg"
    ]
  }
]

export { PRODUCT_MOCKING, PRODUCT_MOCKING_LIST };
