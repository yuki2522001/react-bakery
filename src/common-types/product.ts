import { CategoryProps } from "./category";

export interface Product {
  id: string;
  name: string;
  price: number;
  images: string[];
  quantity: number;
  categoryId: string;
  description: string;
  product?: {};
}

export interface ProductContext {
  products: Product[];
  dispatch: Function,
  searchValue: string;
  setSearchValue: Function;
  categories: CategoryProps[];
  setCategories: Function;
}