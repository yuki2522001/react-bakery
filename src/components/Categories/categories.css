.categories {
  margin-top: 50px;
  text-align: center;
}

.categories__title {
  text-transform: uppercase;
  font-weight: bold;
}

.categories__list {
  display: flex;
  justify-content: center;
  margin: 30px 0 40px 0;
  padding: 0;
}

.categories__item {
  list-style: none;
  cursor: pointer;
  margin: 0 30px;
  position: relative;
  color: var(--text-secondary);
}

.categories__item:hover {
  color: var(--text-primary);
}

.active {
  color: var(--text-primary);
}

.categories__item > a {
  color: var(--text-third);
  text-decoration: none;
}