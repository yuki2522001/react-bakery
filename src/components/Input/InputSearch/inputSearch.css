.search {
  display: flex;
  align-items: center;
}

.search__text {
  width: 250px;
  padding: 10px;
  border: none;
}