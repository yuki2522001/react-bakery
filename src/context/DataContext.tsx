import { ProductContext } from "@common-types/product";
import { createContext, useMemo, useReducer, useState } from "react";
import { productReducer } from "@reducer/productReducer";
import { DataState } from "@common-types/data";
import { CategoryProps } from "@common-types/category";

const initialState: DataState = {
  products: [],
};

const DataContext = createContext<ProductContext>({} as ProductContext);
const DataProvider: React.FC<{children: JSX.Element[] | JSX.Element}> = ({ children }) => {
  const [searchValue, setSearchValue] = useState<string>("");
  const [categories, setCategories] = useState<CategoryProps[]>([]);

  const [state, dispatch] = useReducer(productReducer, initialState);
  const { products } = state;

  const value = useMemo(() => ({
    products,
    dispatch,
    searchValue,
    setSearchValue,
    categories,
    setCategories,
  }), [products, searchValue]);

  return (
    <DataContext.Provider value={value}>
      {children}
    </DataContext.Provider>
  )
}

export {DataProvider, DataContext};