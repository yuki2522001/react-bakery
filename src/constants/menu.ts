import { MenuType } from "../common-types/menu";

const MENU_LIST: MenuType[] = [
  {
    key: "about",
    label: "About",
  },
  {
    key: "shop",
    label: "Shop",
  },
  {
    key: "blog",
    label: "Blog",
  },
  {
    key: "faq",
    label: "FAQ",
  },
  {
    key: "contact",
    label: "Contact"
  }
];

export default MENU_LIST;
