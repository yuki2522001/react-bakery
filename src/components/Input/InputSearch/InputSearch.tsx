import React, { useContext } from "react";
import { DataContext } from "@context/DataContext";
import "./inputSearch.css";

const InputSearch: React.FC = () => {
  // handle search with SearchContext
  const { setSearchValue } = useContext(DataContext);
  const handleSearch = (e: { target: { value: string } }) => {
    // get current name
    const nameLike = { name_like: e.target.value };
    setSearchValue?.(nameLike);
  };

  return (
    <div data-testid="input-search" className="search">
      <input
        className="search__text"
        type="text"
        placeholder="Search item"
        onChange={handleSearch}
      />
    </div>
  );
};

export default InputSearch;
