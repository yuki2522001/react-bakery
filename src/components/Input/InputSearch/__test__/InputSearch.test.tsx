import { act, fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import InputSearch from "../InputSearch";
import { DataContext } from "@context/DataContext";
import { ProductContext } from "@common-types/product";
import { PRODUCT_MOCKING_LIST } from "@__mocks__/constants/product";
import { CATEGORY_MOCKING_LIST } from "@__mocks__/constants/categories";

const contextProductMock: ProductContext = {
  products: PRODUCT_MOCKING_LIST,
  dispatch: jest.fn(),
  searchValue: "",
  setSearchValue: jest.fn(),
  categories: CATEGORY_MOCKING_LIST,
  setCategories: jest.fn()
};

describe("InputSearch component", () => {
  test("should render input search component", () => {
    const { getByTestId } = render(<InputSearch />);
    expect(getByTestId("input-search")).toBeInTheDocument();
  });

  test("display product after inputSearch", async () => {
    act(() => {
      render(
        <DataContext.Provider value={contextProductMock}>
          <InputSearch />
        </DataContext.Provider>
      );
      const inputElement = screen.getByPlaceholderText(
        /Search item/i
      ) as HTMLInputElement;
      fireEvent.change(inputElement, { target: { value: "Cheese pocket" } });
      expect(inputElement.value).toBe("Cheese pocket");
    });
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<InputSearch />);
    expect(asFragment()).toMatchSnapshot();
  });
});
