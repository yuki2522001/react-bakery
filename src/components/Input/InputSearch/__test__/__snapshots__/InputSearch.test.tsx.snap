// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSearch component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="search"
    data-testid="input-search"
  >
    <input
      class="search__text"
      placeholder="Search item"
      type="text"
    />
  </div>
</DocumentFragment>
`;
